app/views.py

```diff
@@ -0,0 +1,102 @@
+from flask import current_app, Blueprint, render_template, abort, request, url_for, flash, redirect
+from jinja2 import TemplateNotFound
+from .forms import AddForm
+from sqlalchemy import create_engine, Column, Integer, String, func
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+main_page = Blueprint('mainpage ', __name__)
+
+# init database
+engine = create_engine('sqlite:///books.db', echo=False)
+Base = declarative_base()
+Session = sessionmaker(bind=engine)
+session = Session()
+
+class Book(Base):
+    __tablename__ = 'books'
+
+    id = Column(Integer, primary_key=True)
+    name = Column(String)
+    author = Column(String)
+    pages = Column(Integer)
+
+    def __repr__(self):
+        return "<Book(name='%s', author='%s', pages='%s')>" % (
+                            self.name, self.author, self.pages)
+
+Base.metadata.create_all(engine)
+
+def update_books(book, action=''):
+    if action == 'delete':
+        session.delete(book)
+    elif action == 'add':
+        session.add(book)
+    session.commit()

```

Comment:
Does flask-sqlachemy have hooks to commit/rollback? Normally you don't commit your sessions in the WSGI apps, because there's a request teardown phase where you either commit either rollback on the error.